code/game/objects/structures/watercloset.dm

```diff
@@ -109,19 +143,30 @@
 				return
 			w_items += I.w_class
 			to_chat(user, "<span class='notice'>You carefully place [I] into the cistern.</span>")
-
+			return
+	else if(istype(I, /obj/item/paper))
+		centrifuge = TRUE
+		to_chat(user, "<span class='notice'>You stick some fitering papers inside</span>")
+		qdel(I)
+		return TRUE
 	else if(istype(I, /obj/item/reagent_containers))
 		if (!open)
 			return
 		if(istype(I, /obj/item/reagent_containers/food/snacks/monkeycube))
 			var/obj/item/reagent_containers/food/snacks/monkeycube/cube = I
 			cube.Expand()
 			return
-		var/obj/item/reagent_containers/RG = I
-		RG.reagents.add_reagent(/datum/reagent/water, min(RG.volume - RG.reagents.total_volume, RG.amount_per_transfer_from_this))
-		to_chat(user, "<span class='notice'>You fill [RG] from [src]. Gross.</span>")
-	else
-		return ..()
+		if(!output && centrifuge)
+			if(user.transferItemToLoc(I, src))
+				output = I
+				to_chat(user, "<span class='notice'>You insert an output beaker.</span>")
+				return TRUE
+	return ..()
```

Comment:
why change this from else and then add all the returns above?

Reply:
its easier to read

Reply:
so code churn